care/facility/tasks/asset_monitor.py

```diff
@@ -31,9 +31,12 @@ def check_asset_status():
             continue
         try:
             # Fetching middleware hostname
-            hostname = asset.meta.get(
-                "middleware_hostname",
-                asset.current_location.facility.middleware_address,
+            hostname = (
+                asset.meta.get(
+                    "middleware_hostname",
+                    asset.current_location.middleware_address,
+                )
+                or asset.current_location.facility.middleware_address
             )
             result: Any = None
 
```

Comment:
We should add a condition to check if hostname is none and skip the AssetClass initialization

Reply:
Added the condition.